src/Microsoft.SqlTools.ServiceLayer/Formatter/TSqlFormatterService.cs

```diff
@@ -0,0 +1,270 @@
+
+                if (settings.PlaceSelectStatementReferencesOnNewLine.HasValue) { options.PlaceEachReferenceOnNewLineInQueryStatements = settings.PlaceSelectStatementReferencesOnNewLine.Value; }
+
+                if (settings.UseBracketForIdentifiers.HasValue) { options.EncloseIdentifiersInSquareBrackets = settings.UseBracketForIdentifiers.Value; }
+
+                options.DatatypeCasing = settings.DatatypeCasing;
+                options.KeywordCasing = settings.KeywordCasing;
+            }
+            return options;
+        }
+        
+        private ScriptFile GetFile(DocumentFormattingParams docFormatParams)
+        {
+            if (WorkspaceService == null)
+            {
+                // TODO throw
+            }
+            return WorkspaceService.Workspace.GetFile(docFormatParams.TextDocument.Uri);
+        }
+
+        private static TextEdit PrepareEdit(ScriptFile scriptFile)
+        {
+            Position start = new Position { Line = 0, Character = 0 };
+            int lastChar = scriptFile.FileLines[scriptFile.FileLines.Count - 1].Length;
+            Position end = new Position { Line = scriptFile.FileLines.Count -1, Character = lastChar };
+
+            TextEdit edit = new TextEdit
+            {
+                Range = new Range { Start = start, End = end }
+            };
+            return edit;
+        }
+
+        // TODO Assess whether to return a single TextEdit for the formatting, or multiple edits.
+        //private static TextEdit[] ConvertToEdits(ScriptFile scriptFile, ReplacementQueue replacements)
+        //{
+        //    List<TextEdit> edits = new List<TextEdit>();
+        //    foreach (Replacement replacement in replacements)
+        //    {
+        //        BufferRange bufferRange = scriptFile.GetRangeBetweenOffsets(replacement.StartIndex, replacement.EndIndex);
+        //        var textEdit = new TextEdit()
+        //        {
+        //            NewText = replacement.NewValue,
+        //            Range = new Range
+        //            {
+        //                Start = new Position { Line = bufferRange.Start.Line, Character = bufferRange.Start.Column },
+        //                End = new Position { Line = bufferRange.End.Line, Character = bufferRange.End.Column }
+        //            }
+        //        };
+        //        edits.Add(textEdit);
+        //    }
+        //    return edits.ToArray();
+        //}
+
+        private async Task HandleRequest<T>(Func<Task<T>> handler, RequestContext<T> requestContext, string requestType)
+        {
+            Logger.Write(LogLevel.Verbose, requestType);
+
+            try
+            {
+                T result = await handler();
+                await requestContext.SendResult(result);
+            }
+            catch (Exception ex)
+            {
+                await requestContext.SendError(ex.ToString());
+            }
+        }
+
+
+
+        public string Format(TextReader input)
```

Comment:
No - all rely on the DoFormat method which uses ServiceProvider to get the Formatter factories. Therefore you need these to be instance-level methods since they require the ServiceProvider to be available. 

---
In reply to: [101188584](https://github.com/Microsoft/sqltoolsservice/pull/229#discussion_r101188584) [](ancestors = 101188584)